src/compiler/checker.ts

```diff
@@ -24437,16 +24438,29 @@ namespace ts {
             if (isInJSFile(node) && getAssignedExpandoInitializer(node)) {
                 return checkExpression(node.right, checkMode);
             }
+            checkGrammarNullishCoalesceWithLogicalExpression(node);
             return checkBinaryLikeExpression(node.left, node.operatorToken, node.right, checkMode, node);
         }
 
+        function checkGrammarNullishCoalesceWithLogicalExpression (node: BinaryExpression) {
+            const { left, operatorToken, right } = node;
+            if (operatorToken.kind === SyntaxKind.QuestionQuestionToken) {
+                if (isBinaryExpression(left) && (left.operatorToken.kind === SyntaxKind.BarBarToken || left.operatorToken.kind === SyntaxKind.AmpersandAmpersandToken)) {
+                    error(left, Diagnostics.Operator_0_cannot_immediately_contain_or_be_contained_within_an_1_operation, tokenToString(left.operatorToken.kind), tokenToString(operatorToken.kind));
+                }
+                if (isBinaryExpression(right) && (right.operatorToken.kind === SyntaxKind.BarBarToken || right.operatorToken.kind === SyntaxKind.AmpersandAmpersandToken)) {
+                    error(right, Diagnostics.Operator_0_cannot_immediately_contain_or_be_contained_within_an_1_operation, tokenToString(right.operatorToken.kind), tokenToString(operatorToken.kind));

```

Comment:
@DanielRosenwasser  Is this enough or should we add did you mean to parenthesize the expression? Do we need code fix for that?

Reply:
Also i am not sure if this is suppose to be error? Where does it say its error?

Reply:
I think it should be

```
'{0}' and '{1}' operations cannot be mixed without parentheses.
```

Quick fixes would be lovely, but aren't required for this iteration.

* `Rewrite as '(a {0} b) {1} c'.`
* `Rewrite as 'a {0} (b {1} c)'.`

If you introduce quick fixes, please ensure that they have differentiated quick fix IDs so we can tell when `??` is the operation people intend to occur first.

Reply:
> Also i am not sure if this is suppose to be error? Where does it say its error?

The grammar doesn't make it clear (it used to when it was a grammar flag), but you can see it on https://tc39.es/proposal-nullish-coalescing/#sec-intro

![image](https://user-images.githubusercontent.com/972891/64378096-c8ccce00-cfe0-11e9-8036-c8318361e223.png)